### PR TITLE
Update setup for developer-centric scripts

### PR DESCRIPTION
**Pull request by razvand**
Update setup to include scripts for building and running. Restructure contents to use the `workdir/` directory as the directory to store repository clones and build output; use `defconfigs/` to store default configuration files.

This relies on [`lib-wamr` #9 PR](https://github.com/unikraft/lib-wamr/pull/9).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,150 +130,345 @@ The following repositories are required for wamr:
...
      ```console
-     $ git clone https://github.com/unikraft/unikraft unikraft
+     tree -F -L 2 workdir/
      ```
 
-  4. While inside the `.unikraft/` directory, create the `libs/` directory:
-
-     ```console
-     $ mkdir libs
+     The layout of the `workdir/` directory should look something like this:
```
Comment (StefanJum) [inline README.md:165]:
```suggestion
```console
     The layout of the `workdir/` directory should look something like this:
```

File: README.md [hunk 2]
```diff
@@ -51,150 +130,345 @@ The following repositories are required for wamr:
...
+cat scripts/run/fc-x86_64-initrd-interp.sh
+```
 
-## Configure
+Now, invoke each script to build and run the application.
+A sample build and run set of commands is:
+
+```console
+./scripts/build/make-qemu-x86_64-9pfs.sh
+./scripts/run/qemu-x86_64-9pfs-interp.sh
```
Comment (StefanJum) [inline README.md]:
```suggestion
./scripts/run/qemu-x86_64-9pfs.sh
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
See the comments.
Besides those, the main file does not load properly.
It gives a `magic header not detected` error on initrd and a `did not select --repl nor -initrd mode!` warning on 9pfs.

The way `lib-wamr` works now, it expects the actual main file (not a cpio archive) to be passed as an initrd, without mounting it as a filesystem. So it would be `CONFIG_LIBVFSCORE_AUTOMOUNT_ROOTFS=n` in the config files, and the run command should be `qemu-system-x86_64 -kernel build/wamr_qemu-x86_64 -initrd main.wasm  -nographic`.

This means the `9pfs`-`initrd` separation is no longer relevant. Also this might not work with the `generate.py` script, we might want to add some way of configuring extra arguments to be passed to the `qemu` run command.

**Comment by razvand** [comment]
@StefanJum , this relies on [`lib-wamr` PR #9](https://github.com/unikraft/lib-wamr/pull/9).

**Comment by StefanJum** [comment]
@razvand does [`lib-wamr`#9](https://github.com/unikraft/lib-wamr/pull/9) works for you? I don't get the errors anymore but the file is not executed. The application just closes after printing the Unikraft banner.
I can look into it, just want to make sure if it's something wrong on my config.

**Comment by razvand** [comment]
> @razvand does [`lib-wamr`#9](https://github.com/unikraft/lib-wamr/pull/9) works for you? I don't get the errors anymore but the file is not executed. The application just closes after printing the Unikraft banner. I can look into it, just want to make sure if it's something wrong on my config.

lemme retry

**Comment by razvand** [comment]
> > @razvand does [`lib-wamr`#9](https://github.com/unikraft/lib-wamr/pull/9) works for you? I don't get the errors anymore but the file is not executed. The application just closes after printing the Unikraft banner. I can look into it, just want to make sure if it's something wrong on my config.
> 
> lemme retry

@StefanJum , I've retried. I used `lib-wamr` in `workdir/libs/` checked out at PR #9. And I tested both with initrd and 9pfs. It all works:

```console
./scripts/generate.py
./scripts/build/make-qemu-x86_64-9pfs.sh
./scripts/run/qemu-x86_64-9pfs.sh
./scripts/build/make-qemu-x86_64-initrd.sh
./scripts/run/qemu-x86_64-initrd.sh
```

**Review by StefanJum — APPROVED** [review]
All good, thanks.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Pull request merged**